scripts/parseDownloadRequests.py

```diff
@@ -0,0 +1,89 @@
+import os
+import boto3
+import re
+import numpy 
+
+from model import Edition, Item, Link
+from model.postgres.item import ITEM_LINKS
+from managers import DBManager
+
+s3_client = boto3.client("s3")
+
+bucketName = 'ump-pdf-repository-logs'
+
+def main():
+    
+    
+    '''
+    The edition title, identifier, and timestamp are parsed out of the 
+    S3 server access log files for UMP download requests
+    '''
+
+    requestRegex = r'REST.GET.OBJECT '
+    fileIDRegex = r'REST.GET.OBJECT (.+pdf\s)' #File ID includes the file name for the pdf object
+    timeStampRegex = r'\[.+\]'
+    referrerRegex = r'https://drb-qa.nypl.org/'
+    umpDownloadArray = [['title', 'timeStamp', 'identifier']]
+
+    batches = load_batch()
+    for batch in batches:
+        for c in batch['Contents']:
+            currKey = str(c['Key'])
+            #logObject is a dict type
+            logObject = s3_client.get_object(Bucket= bucketName, Key= f'{currKey}')
+            for i in logObject['Body'].iter_lines():
+                logObject = i.decode('utf8')
+                parseTuple = parseInfo(logObject, requestRegex, referrerRegex, timeStampRegex, fileIDRegex)
+                if parseTuple:
+                    umpDownloadArray.append(parseTuple)
+    umpDownloadCSV = numpy.array(umpDownloadArray)
+    with open('data3.csv', 'w') as f: 
+        f.write(str(umpDownloadCSV))
+
+def load_batch():
+    paginator = s3_client.get_paginator('list_objects_v2')
+    page_iterator = paginator.paginate(Bucket= bucketName, Prefix= 'logs/946183545209/us-east-1/ump-pdf-repository/2024/03/13/')
```

Comment:
Can we set the file name and the log prefix as variables at the top of the page (along with the regex constants)? I imagine even if we run this manually we might want an easy way to change the output file names and prefixes, for example.

Reply:
That works for me. Makes sense to have these variable be easy to change on a global scale in the script.